app/scripts/modules/core/src/serverGroup/serverGroupSearchResultType.tsx

```diff
@@ -46,22 +54,127 @@ const SearchResultsHeader: SearchResultsHeaderComponent = () => (
+  return sgToFetch.map(toFetch => ({ toFetch, fetched: findFetchedValue(toFetch) }));
+};
+
+const fetchServerGroups = (toFetch: IServerGroupSearchResult[]): Observable<IServerGroupTuple[]> => {
+  const fetchPromise = ReactInjector.API.one('serverGroups')
+    .withParams({ serverGroupNames: toFetch.map(sg => `${sg.account}:${sg.region}:${sg.serverGroup}`) })
+    .get()
+    .then((fetched: IServerGroup[]) => makeServerGroupTuples(toFetch, fetched));
+
+  return Observable.fromPromise(fetchPromise);
+};
+
+/**
+ * HOC which provides provides server group health/instance counts
+ *
+ * The /search endpoint does not return instance counts.
+ * This component fetches instance counts separately and provides them to the wrapped component.
+ * This component waits until it is rendered, then starts fetching instance counts.
+ * It mutates the input search results and then passes the data to the nested component.
+ */
+const AddHealthCounts = (Component: SearchResultsDataComponent<IServerGroupSearchResult>): SearchResultsDataComponent<IServerGroupSearchResult> => {
+  return class FetchHealthCounts extends React.Component<IServerGroupDataProps, IServerGroupDataState> {
+    public state = { serverGroups: [] } as any;
+    private results$ = new BehaviorSubject<IServerGroupSearchResult[]>([]);
+    private stop$ = new Subject();
+
+    constructor(props: any) {
+      super(props);
+
+      // results to use when a fetch has failed.
+      const failedFetch = (failedFetches: IServerGroupSearchResult[]) =>
+        Observable.of(failedFetches.map(toFetch => ({ toFetch, fetched: { instanceCounts: null } as IServerGroup })));
+
+      // fetch a batch of server groups.
+      const processBatch = (batch: IServerGroupSearchResult[]): Observable<IServerGroupTuple[]> => {
+        return fetchServerGroups(batch).catch((err: { status: number }) => {
+          // In case of 404 during batch fetch, fall back to individual fetch
+          if (err.status === 404) {
+            // retry, but fetch each servergroup individually
+            return Observable.from(batch).mergeMap(sg => {
+              return fetchServerGroups([sg]).catch(() => failedFetch([sg]))
+            });
+          }
+
+          return Observable.of(failedFetch(batch))
+        });
+      };
+
+      this.results$.mergeMap((searchResults: IServerGroupSearchResult[]) => {
+        return Observable.from(searchResults)
+          .filter(result => result.instanceCounts === undefined)
+          // Serially fetch instance counts in batches of 25
+          .bufferCount(25)
+          .concatMap(processBatch);
+      })
+      .takeUntil(this.stop$)
+      .subscribe((tuples: IServerGroupTuple[]) => {
+        tuples.forEach(result => result.toFetch.instanceCounts = result.fetched.instanceCounts);
+        this.setState({ serverGroups: this.results$.value.slice() });
+      });
+    }
```

Comment:
I really like this approach - nice work figuring out the RxJS to do this cleanly.